### PR TITLE
Replace swagger ui with redoc

### PR DESCRIPTION
**Pull request by jcalcaben**
## This PR is a:

- [x] New feature
- [ ] New topic
- [ ] Content update
- [ ] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged, it will replace the swagger UI with ReDoc in 2.3.

### CODE REVIEW

**Comment by jeff-matthews** [comment]
I was hoping to get more internal feedback on this before implementing, but no one has offered any. I think we should merge this. @keharper, do you have any outstanding concerns with doing this for 2.3?

**Comment by keharper** [comment]
I have not had time to give this a thorough review. This is not a 15 minute review you're asking for. 

A few obvious omissions:

* The schemas do not contain Commerce, B2B, or MSI endpoints. You could copy the latest schema from the swagger directory, but that won't help with the customer-specific schema. Guest is probably unaffected.
* The 2.3 index page still points to Swagger. There are other places that I refer to generated documentation that needs review. 
* The redoc index page needs links to the topics that describe how to generate tokens

**Comment by jeff-matthews** [comment]
Understood @keharper. We'll wait on merging this until you have time early next week to review.

**Comment by jeff-matthews** [comment]
Thanks for adding those updates @jcalcaben. 

@keharper, you have until Tuesday to provide review feedback. We're going to merge this week.

File: _data/toc/rest-api.yml [hunk 1]
```diff
@@ -17,9 +17,23 @@ pages:
...
 
-        - label: REST Endpoints (Swagger)
+        - label: REST Endpoints (ReDoc)
           include_versions: ["2.3"]
-          url: http://devdocs.magento.com/swagger/index_23.html
+          url: /redoc/2.3/
+          versionless: true
+          children:
+
+            - label: Guest REST endpoints
```
Comment (keharper) [inline _data/toc/rest-api.yml]: Place these in Admin/Customer/Guest order

File: guides/v2.3/rest/tutorials/index.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Before you begin any tutorial, make sure you know the basics about {{site.data.v
 
 * Know how to construct a REST call in Magento. See [Construct a request]({{ page.baseurl }}/get-started/gs-web-api-request.html) for details.
 
-* Find the Magento REST API documentation. You can view the [static REST API documentation on devdocs]({{ site.baseurl }}/swagger/){:target="_blank"} or [generate a local API reference]({{ page.baseurl }}/rest/generate-local.html).
+* Find the Magento REST API documentation. You can view the [static REST API documentation on devdocs]({{site.baseurl}}/redoc/{{page.guide_version}}/) or [generate a local API reference]({{ page.baseurl }}/rest/generate-local.html).
 
 * Find the Magento Merchant documentation. Refer to [Getting Started with {{site.data.var.ce}}](http://docs.magento.com/m2/ce/user_guide/getting-started.html){:target="_blank"} for information about the Luma store that is created when you install Magento with the sample data.
```
Comment (keharper) [inline guides/v2.3/rest/tutorials/index.md]: Remove {:target="_blank"}

File: redoc/2.3/index.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+Magento exposes different REST endpoints depending on the type of user making the requests.
+You must request and include a security token to gain access to the Customer and Admin REST API endpoints.
+
+See [Token-based authentication][] for instructions on requesting and using security tokens.
+
+Each of the following links lead to a list of REST endpoints specific to a user type:
+
+* [Guest REST API endpoints]({{site.baseurl}}/redoc/2.3/guest-rest-api.html) - Available for anonymous users
```
Comment (keharper) [inline redoc/2.3/index.md]: Place these in Admin > Customer > Guest order

**Comment by keharper** [comment]
I can't comment in the two order tutorial files where you broke the symlink, so I'm adding this comment here:

Since you broke the symlink, delete the v2.1 and v2.2 redirects in the frontmatter and delete the v2.3 redirect in the 2.2 version of these files (`order-intro.md` and `order-issue-refund.md`)

**Pull request merged**